### ISSUE TITLE
Federation with other platforms does not work properly

### ISSUE THREAD

**Issue by Ravenbird** [friendica/friendica#12590]
**Friendica Version:** 2022.12
**Database Version:** 1502/1502
**Post-Update Version:** 1484/1484.

Federation of individual comments via ActivityPub does not work correctly or is incomplete. I noticed this, because in comments were replied to other comments that were not displayed to me at all.

As an example, here is a post with comments that I can see on Friendica:

![](https://picr.eu/images/2023/01/01/M9rD9.jpg)

However, on the instance where the post was posted it looks like this:

![](https://picr.eu/images/2023/01/01/M9LLp.jpg)

That is, various comments were not federated. This makes exchange and discussion across the boundaries of the platforms difficult or impossible.

**Comment by utzer** [comment]
I see the same behavior and remember that in the past the threads were completed on an interaction (like|reshare|comment).

**Comment by Ravenbird** [comment]
There is also a discussion about the problem in german: https://anonsys.net/display/bf69967c-1463-b14c-fb09-bcf319321670

**Comment by MrPetovan** [comment]
Threads have never been perfectly completed, meaning you aren’t supposed to see replies from authors you are not following or one of your follows didn’t interact with.

There has been recent issues with actual thread completion, but nothing in this case seems to hint at this.

If you want a perfectly distributed thread view, Diaspora is better-suited for this expectation. You can even keep all your Friendica contacts!

**Comment by SpcCw** [comment]
Currently it works like this: to see a post or comment your server has to receive it first. This happens if any of these are true:

- you or someone else on your server is subscribed to the author of this item
- the item is reshared by someone you or someone else on your server is subscribed to
- you interacted with child item (e.g. reply to the post) which causes parent items to get fetched

If you just stumbled upon someone's post and there are some comments from people who never made it to your server you won't see them - and I think this is the case for any Fedi platform, not just Friendica. This is also the reason you sometimes don't see full "like" counts for others' posts.

**Comment by annando** [comment]
Concerning the `like` counts: Comments are normally distributed to all followers and there is some relaying on some systems, so that comments should distribute. But likes aren't distributed and it seems as if especially Mastodon is sending likes only to the targeted system. So the like counter will vary a lot - and we cannot do anything about this.

**Comment by SpcCw** [comment]
Also on Mastodon sometimes people leave comments with restricted visibility (they have padlock icon). I noticed that even when you have original post you may not get these comments if they are not directed to you.

**Comment by MrPetovan** [comment]
We have a specific issue for these replies: https://github.com/friendica/friendica/issues/12535

**Comment by utzer** [comment]
> Threads have never been perfectly completed, meaning you aren’t supposed to see replies from authors you are not following or one of your follows didn’t interact with.

Yes, but in the past after an interaction in a thread it was completed, so that old and new comments showed up after that interaction. What changed that this does not happen anymore? 
I don't care if it is not perfect, but now it is much less than perfect. 

 
> If you want a perfectly distributed thread view, Diaspora is better-suited for this expectation. You can even keep all your Friendica contacts!

No AP support, so no alternative, this issue here is about deteriorated thread completion for AP posts especially.

> Federation of individual comments via ActivityPub does not work correctly or is incomplete.

So lets agree it worked better in the past and that something changed. 

Edit: Maybe it is not so bad not to receive all the replies, especially if the Fediverse grows much, but then there should be still triggers to complete local thread copies.
